experiments/compositions/composition/internal/controller/composition_controller.go

```diff
@@ -75,34 +79,56 @@ func (r *CompositionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// Grab status for comparision later
+	oldStatus := composition.Status.DeepCopy()
+
+	// Try updating status before returning
+	defer func() {
+		if !reflect.DeepEqual(oldStatus, composition.Status) {
+			if err := r.Client.Status().Update(ctx, &composition); err != nil {
+				logger.Error(err, "unable to update Composition status")
+			}
+		}
+	}()
+
 	logger = logger.WithName(composition.Name).WithName(fmt.Sprintf("%d", composition.Generation))
 
 	logger.Info("Validating Compostion object")
 	if !composition.Validate() {
 		logger.Info("Validation Failed")
-		if err := r.Client.Status().Update(ctx, &composition); err != nil {
-			logger.Error(err, "unable to update Composition status")
-			return ctrl.Result{}, err
-		}
+		return ctrl.Result{}, fmt.Errorf("Validation failed")
 	}
 
 	logger.Info("Processing Composition object")
-	if err := r.runComposition(ctx, composition, logger); err != nil {
+	if err := r.runComposition(ctx, &composition, logger); err != nil {
+		logger.Info("Error processing Composition")
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, nil
 }
 
 func (r *CompositionReconciler) runComposition(
-	ctx context.Context, c compositionv1.Composition, logger logr.Logger,
+	ctx context.Context, c *compositionv1.Composition, logger logr.Logger,
 ) error {
 	var crd extv1.CustomResourceDefinition
 	logger = logger.WithName(c.Spec.InputAPIGroup)
+	c.Status.ClearCondition(compositionv1.Error)
```

Comment:
We should decide whether we want to remove the condition vs setting it to false, I had some feedback recently that removing the condition was less discoverable and made it harder to consume.  Not a blocker for this PR.